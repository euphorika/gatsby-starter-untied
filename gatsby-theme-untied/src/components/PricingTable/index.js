/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

import { Link } from 'gatsby'
import Img from 'gatsby-image'

const PricingTable = ({
  children,
  id,
  headline,
  price,
  imgFluid,
  callToAction,
}) => (
  <section
    id={id}
    sx={{
      mx: ['auto', 3],
      my: 4,
      width: '100%',
      maxWidth: '300px',
      fontSize: 0,
      py: 2,
      border: 'solid 1px',
      borderColor: 'muted',
      scrollSnapAlign: 'start',
      flexShrink: ['0', 'unset'],
      mr: ['50px', '15px'],
      webkitTransformOrigin: 'center center',
      transformOrigin: 'center center',
      webkitTransform: 'scale(1)',
      transform: 'scale(1)',
      webbkitTransition: '-webkit-transform 0.5s',
      transition: 'transform 0.5s',
    }}
  >
    <h3
      sx={{
        m: 0,
        px: 3,
        fontSize: 2,
      }}
    >
      {headline}
    </h3>
    <p
      sx={{
        mb: 2,
        px: 3,
        fontSize: 2,
      }}
    >
      {price}
    </p>
    <Img fluid={imgFluid} alt={headline} />
    <div
      sx={{
        mt: 2,
        px: 3,
      }}
    >
      {children}
    </div>
    <div
      sx={{
        mt: 3,
        px: 3,
      }}
    >
      <Link
        sx={{
          display: 'inline-block',
          width: '100%',
          textAlign: 'center',
          color: 'btnText',
          fontSize: 0,
          fontWeight: 'bold',
          bg: 'btnBg',
          textDecoration: 'none',
          borderRadius: 0,
          p: 2,
        }}
        to={callToAction.link}
      >
        {callToAction.text}
      </Link>
    </div>
  </section>
)

PricingTable.defaultProps = {
  id: null,
}

PricingTable.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node.isRequired,
  headline: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  imgFluid: PropTypes.object.isRequired,
  callToAction: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  }),
}

export default PricingTable
