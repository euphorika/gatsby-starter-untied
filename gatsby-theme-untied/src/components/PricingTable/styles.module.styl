@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

:root
  --section-highlight-color highlight-color

.pricing-table-section
  margin 0 auto
  max-width 300px
  padding 15px
  background-color var(--section-highlight-color)
