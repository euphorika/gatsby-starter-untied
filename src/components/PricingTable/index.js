import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'

import styles from './styles.module.styl'

const PricingTable = ({ children, headline, price, callToAction }) => (
  <section className={styles.pricingTableSection}>
    <h3>{headline}</h3>
    <p>{price}</p>
    <div className={styles.description}>
      {children}
    </div>
    <div className={styles.callToAction}>
      <Link to={callToAction.link}>{callToAction.text}</Link>
    </div>
  </section>
)

PricingTable.propTypes = {
  children: PropTypes.node.isRequired,
  headline: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  callToAction: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired
  })
}

export default PricingTable
