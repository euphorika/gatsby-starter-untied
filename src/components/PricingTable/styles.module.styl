@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.pricing-table-section
  margin 0
