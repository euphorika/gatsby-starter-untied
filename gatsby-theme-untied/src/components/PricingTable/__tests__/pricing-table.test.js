import React from "react"
import renderer from "react-test-renderer"
import PricingTable from "../"

describe("PricingTable", () => {
  it("renders correctly", () => {
    const imgFluid = {
      aspectRatio: 1.0,
      base64: "String",
      sizes: "(max-width: 4928px) 100vw, 4928px",
      src: "image.jpg",
      srcSet: "srcSet.image.jpg",
    }
    const callToAction = {
      text: "Text",
      link: "/",
    }
    const pricingTable = (
      <PricingTable
        headline="headline"
        price="1.22"
        imgFluid={imgFluid}
        callToAction={callToAction}
      >
        Description
      </PricingTable>
    )

    const tree = renderer.create(pricingTable).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
