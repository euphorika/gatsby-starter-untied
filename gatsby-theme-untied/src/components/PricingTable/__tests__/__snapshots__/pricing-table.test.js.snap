// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingTable renders correctly 1`] = `
<section
  className="css-1n7py6i"
  id={null}
>
  <div
    className="css-1gwydv7"
  >
    <h3
      className="css-15mnsi5"
    >
      headline
    </h3>
    <p
      className="css-12l35fi"
    >
      1.22
    </p>
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "100%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        src="String"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 4928px) 100vw, 4928px"
          srcSet="srcSet.image.jpg"
        />
        <img
          alt="headline"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 4928px) 100vw, 4928px"
          src="image.jpg"
          srcSet="srcSet.image.jpg"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"srcSet.image.jpg\\" sizes=\\"(max-width: 4928px) 100vw, 4928px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 4928px) 100vw, 4928px\\" srcset=\\"srcSet.image.jpg\\" src=\\"image.jpg\\" alt=\\"headline\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
    <div
      className="css-tgll2l"
    >
      Description
    </div>
    <div
      className="css-tgll2l"
    >
      <Link
        className="css-1obveo8"
        to="/"
      >
        Text
      </Link>
    </div>
  </div>
</section>
`;
